lib/neovim-plugin.nix

```diff
@@ -161,8 +162,48 @@
                   # Add the plugin setup code `require('foo').setup(...)` to the lua configuration
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
 
-                  # Write the lua configuration `luaConfig.content` to the config file
-                  (setLuaConfig cfg.luaConfig.content)
+                  # Write the lua configuration `luaConfig.content` to the config file when lazy loading is not enabled
+                  (lib.mkIf (!cfg.lazyLoad.enable) (setLuaConfig cfg.luaConfig.content))
+
+                  # When lazy loading is enabled for this plugin, route its configuration to the enabled provider
+                  (lib.mkIf cfg.lazyLoad.enable {
+                    assertions = [
+                      {
+                        assertion = (isColorscheme && colorscheme != null) || cfg.lazyLoad.settings != { };
+                        message = "You have enabled lazy loading for ${originalName} but have not provided any configuration.";
+                      }
+                    ];
+
+                    warnings = lib.optional (!config.plugins.lz-n.enable) ''
+                      You have enabled lazy loading for ${originalName} but have not enabled any lazy loading plugins.
+                      We will not insert the lua for this plugin and you must handle loading it yourself.  
+
+                      Currently supported lazy providers:
+                        - lz-n
+                    '';
+
+                    plugins.lz-n = {
+                      plugins = [
+                        (
+                          {
+                            __unkeyed-1 = originalName;
+                            # Use provided after, otherwise fallback to normal function wrapped lua content
+                            after =
+                              let
+                                after = cfg.lazyLoad.settings.after or null;
+                                default = "function()\n " + cfg.luaConfig.content + " \nend";
+                              in
+                              if (lib.isString after || lib.types.rawLua.check after) then after else default;
+                            colorscheme = lib.mkIf isColorscheme (cfg.lazyLoad.settings.colorscheme or colorscheme);
+                          }
+                          // (lib.removeAttrs cfg.lazyLoad.settings [
+                            "after"
+                            "colorscheme"
+                          ])
```

Comment:
```suggestion
                          // lib.removeAttrs cfg.lazyLoad.settings [
                            "after"
                            "colorscheme"
                          ]
```

Reply:
I believe the new formatting would also permit
```nix
// lib.removeAttrs cfg.lazyLoad.settings [ "after" "colorscheme" ]
```

Although whether that's bettor or worse in this case is up for debate. If there's a chance the list well grow with time, the current formatting is preferable to minimize  noise in future patches.